sqldelight-idea-plugin/src/main/kotlin/app/cash/sqldelight/intellij/ProjectService.kt

```diff
@@ -164,10 +169,21 @@ class ProjectService(val project: Project) : SqlDelightProjectService, Disposabl
 
   override var dialect: SqlDelightDialect = MissingDialect()
 
+  private fun LanguageParserDefinitions.updatePredefinedTables(language: Language, newParser: SqlParserDefinition) {
+    val oldParser = forLanguage(language)
+    removeExplicitExtension(language, oldParser)
+    addExplicitExtension(language, newParser)
```

Comment:
whats the reason for doing this, are the predefined tables being cached somewhere that this removal cleans up?

Reply:
Yes. The ParserDefinitions are instanced and cached by IntelliJ.